src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6693,10 +6693,15 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         }
         this->visit_expr(*x.m_value);
 
-        // If tmp is a statement and not an expression
-        // never cast into expression using ASRUtils::EXPR
-        // Just ignore and exit the function naturally.
-        if( tmp && !ASR::is_a<ASR::stmt_t>(*tmp) ) {
+        if (eval_count > 0) {
+            // In Interactive mode
+            if ((tmp) && (!ASR::is_a<ASR::expr_t>(*tmp))) {
+                tmp = nullptr;
+            }
+        } else if (tmp && !ASR::is_a<ASR::stmt_t>(*tmp)) {
```

Comment:
Can you check if doing the following works?

```suggestion
        if (eval_count == 0 && tmp && !ASR::is_a<ASR::stmt_t>(*tmp)) {
```

Reply:
These changes work. I have committed it.